src/moonlink_datafusion/src/connection_pool.rs

```diff
@@ -0,0 +1,361 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        LazyLock,
+    },
+    time::{Duration, Instant},
+};
+use tokio::{net::UnixStream, sync::Mutex, time::sleep};
+
+struct PooledEntry {
+    stream: UnixStream,
+    last_used: Instant,
```

Comment:
Incorrect variable naming I think, it's the timestamp where the entry is inserted into the pool, not the timestamp it was last used.